internal/graph/cycle_check_resolver_test.go

```diff
@@ -60,3 +66,56 @@ func TestCycleDetectionCheckResolver(t *testing.T) {
 		require.True(t, resp.GetAllowed())
 	})
 }
+
+func TestIntegrationWithLocalChecker(t *testing.T) {
+	t.Cleanup(func() {
+		goleak.VerifyNone(t)
+	})
+
+	cycleDetectionCheckResolver := NewCycleDetectionCheckResolver()
+	t.Cleanup(cycleDetectionCheckResolver.Close)
+	localCheckResolver := NewLocalChecker()
+	t.Cleanup(localCheckResolver.Close)
+
+	cycleDetectionCheckResolver.SetDelegate(localCheckResolver)
+	localCheckResolver.SetDelegate(cycleDetectionCheckResolver)
+
+	ds := memory.New()
+	t.Cleanup(ds.Close)
+
+	storeID := ulid.Make().String()
+
+	model := parser.MustTransformDSLToProto(`
+		model
+		  schema 1.1
+
+		type user
+
+		type group
+		  relations
+			define blocked: [user, group#member]
+			define member: [user, group#member] but not blocked
+`)
+
+	err := ds.Write(context.Background(), storeID, nil, []*openfgav1.TupleKey{
+		tuple.NewTupleKey("group:1", "member", "user:jon"),
+		tuple.NewTupleKey("group:1", "blocked", "group:1#member"),
+	})
+	require.NoError(t, err)
+
+	typesys, err := typesystem.NewAndValidate(
+		context.Background(),
+		model,
+	)
+	require.NoError(t, err)
+
+	ctx := storage.ContextWithRelationshipTupleReader(context.Background(), ds)
+	ctx = typesystem.ContextWithTypesystem(ctx, typesys)
+	_, err = cycleDetectionCheckResolver.ResolveCheck(ctx, &ResolveCheckRequest{
+		StoreID:              storeID,
+		AuthorizationModelID: model.GetId(),
+		TupleKey:             tuple.NewTupleKey("group:1", "blocked", "user:jon"),
+		RequestMetadata:      NewCheckRequestMetadata(25),
+	})
+	require.NoError(t, err)
```

Comment:
```suggestion
	require.NoError(t, err)
	require.False(t, resp.GetAllowed())
```

Reply:
I get `allowed=true` 😬

Reply:
Wait a second, this should be an error `ErrCycleDetected`. Sorry, I stand corrected.

If you start at `group#blocked` --> `group#member` and `group#member` depends on `group#blocked`, then that's a cycle and you cannot resolve. 

So this should be an `ErrCycleDetected`.

Reply:
should we tackle this in a different PR?